### PR TITLE
OOP school library

### PR DESCRIPTION
**Pull request by Isabellanamutenda**
OOP school library
In this project I:

- Created a class classroom with its requirements
- Created a class book
- Add a relationship between book and person
- Created has-many/belongs-to relationship between person and book

### CODE REVIEW

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Ashley, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

It looks like you submitted a review request for the wrong milestone, the milestone you finished is "set up associations" and the one you requested a review for is "add basic UI"

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (SajjadAhmad14) [inline book.rb]: Please create a new instance of the Rental class in `#add_rental` here.

```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
```
Comment (SajjadAhmad14) [inline book.rb:9]: `#add_rental` of `Person` class should receive `book, date` as parameters.
```
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Ashley, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
README is professional. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: book.rb (hunk 2 above)
Comment (Isabellanamutenda) [reply book.rb:9]: Fixed

File: classroom.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @student = []
+  end
+
+  def add_student(student)
+    @students << student
+    student.classroom = self
```
Comment (D4nielJ) [inline classroom.rb:12]: [OPTIONAL] I think you might have a typo here. `@student` instead of `@students` in `initialize`

**Review by D4nielJ — APPROVED** [review]
## Hello @Isabellanamutenda, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**